tempo_embeddings/embeddings/vector_database.py

```diff
@@ -0,0 +1,336 @@
+        return self._batch_size
+
+    @batch_size.setter
+    def batch_size(self, batch_size: int) -> None:
+        self._batch_size = batch_size
+
+
+    def _batches(self, passages: list[Passage]) -> Iterable[list[Passage]]:
+        for batch_start in tqdm(
+            range(0, len(passages), self.batch_size),
+            desc="Embeddings Batches",
+            unit="batch",
+            total=len(passages) // self.batch_size + 1,
+        ):
+            yield passages[batch_start : batch_start + self.batch_size]
+
+    @abstractmethod
+    def connect(self):
+        pass
+
+    @abstractmethod
+    def insert_passages_embeddings(self, collection:Collection, passages: List[Passage]):
+        pass
+
+    @abstractmethod
+    def retrieve_vectors_if_exist(self, collection: Collection, passages: List[Passage]):
+        pass
+
+
+class ChromaDatabaseManager(VectorDatabaseManagerWrapper):
+    """A Chroma Database can create N collections and will always use THE SAME embedder function and tokenizer for all collections.
+        To create collections with different embedders one needs to create separate databases
+    """
+    def __init__(self, 
+                 db_path: str = "default_db",
+                 embedder_name: str = "sentence-transformers/all-MiniLM-L6-v2",
+                 embedder_config: Dict[str,Any] = None,
+                 batch_size: int = 8):
+        super().__init__(batch_size)
+        self.db_path = db_path
+        self.embedder_name = embedder_name
+        self.embedder_config = embedder_config
+        self.tokenizer = AutoTokenizer.from_pretrained(embedder_name)
+        self.client =  None
+        self.requires_explicit_embeddings = False
+        if self.embedder_config and self.embedder_config.get("type") == "hf":
+            self.embedding_function = embedding_functions.HuggingFaceEmbeddingFunction(
+                api_key=self.embedder_config.get("api_key"),
+                model_name=embedder_name
+            )
+        elif self.embedder_config and self.embedder_config.get("type") == "custom":
+            self.requires_explicit_embeddings = True
+            self.embedding_function = None
+        else:
+            self.embedding_function = None
+
+        self.config = {
+            "embedder_name": self.embedder_name,
+            "embedder_type": self.embedder_config["type"],
+            "requires_explicit_embeddings": self.requires_explicit_embeddings,
+            "existing_collections": []
+        }
+    
+    def _save_config(self):
+        if self.config:
+            json.dump(self.config, open(f"{self.db_path}/config.json", "w"), indent=2)
+
+    def connect(self):
+        if not self.client:
+            self.client = chromadb.PersistentClient(path=self.db_path)
+            # If the given path existed then we will load the records AND override the parameters with the config
+            config_path = f"{self.db_path}/config.json"
+            if os.path.exists(config_path):
```

Comment:
Does that mean that we ignore the configuration that was passed to the `__init__()` method? If so, I think there should at least be a warning, perhaps even an exception.